src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesReconTest.kt

```diff
@@ -237,7 +254,138 @@ class ActivitiesReconTest {
         .map { DpsBookingCount(it.bookingId, it.dpsCount!!) }
         .also {
           whenever(activitiesApiService.getAllocationReconciliation(anyString()))
-            .thenReturn(DpsResponse(prisonId, it))
+            .thenReturn(DpsAllocationResponse(prisonId, it))
+        }
+
+      // stub the booking details calls for all will be reported on (that don't match counts)
+      bookingCounts
+        .filterNot { it.nomisCount == it.dpsCount }
+        .map { PrisonerDetails(it.offenderNo, it.bookingId, it.location) }
+        .also { whenever(nomisApiService.getPrisonerDetails(any())).thenReturn(it) }
+    }
+  }
+
+  @Nested
+  inner class AttendanceReconciliationReport {
```

Comment:
Don't need as many tests as for allocations - reusing some of the allocation code.